### PR TITLE
Clean up dactyl manuform unused files

### PR DESCRIPTION
**Pull request by DuMaM**
Remove unnecessary files which confuse more and more people. 

## Description
I noticed that beginners don't know how to work with dactyl keyboard, and they don't fully understand project structure.
This PR will make this keyboard code base closer to other well maintained keyboards.

I tested this with my keyboard 6x6 and it works fine. This change is the same for all other variants, so I don't think there is need for any additional tests

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR
None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Honesty, I'm pretty opposed to this.  while I definitely understand the reasoning behind it, and I understand that it's lower impact for the board, I feel that having it as a good example of how to set up multiple revision boards outweighs the downsides.

**Comment by DuMaM** [comment]
I this solution was used when there was rules.mk which allowed for building whole folder.
Now it just exists there and do nothing particularly useful. 
Mosty because each subfolder is not a new revision but separate keyboard, whit different config, features enabled etc.
The only thing they share together is a config from main folder and guidelines how to wire them, so i don't see a point in holding this files.
There are much better examples how to do proper revision in qmk.

**Comment by DuMaM** [comment]
@drashna  So there is no chance to merge this?

**Comment by drashna** [comment]
For this, I'll defer to other collaborators judgement.

**Comment by DuMaM** [comment]
So I guess owners do not want this change

File: keyboards/handwired/dactyl_manuform/4x5/keymaps/dvorak/keymap.c
Comment (fauxpark) [inline keyboards/handwired/dactyl_manuform/4x5/keymaps/dvorak/keymap.c:3]:
```suggestion
#include QMK_KEYBOARD_H
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by fauxpark** [comment]
Merge conflict needs resolving.

TBH, I'm not a fan of the ifdef chain in these kinds of boards, so I'm good with removing it. The Planck example is maybe not so apt, given it does the same thing.

As long as the root config.h is still picked up, and there aren't any foreseeable "common" items to all Dactyl revisions that necessitate the root `.c`/`.h`, 👍

**Comment by DuMaM** [comment]
@fauxpark done

**Review by drashna — APPROVED** [review]

**Comment by DuMaM** [comment]
bump

**Review by fauxpark — APPROVED** [review]

**Pull request merged**
